### PR TITLE
Configurable zone reload interval in file plugin

### LINKED ISSUE #2108: Configurable reload time of zone files
Zone file reload is currently "hardcoded" to 1 minute https://github.com/coredns/coredns/blob/master/plugin/file/reload.go#L9
In some cases (like more complex provisioning) this is too long time. This can be also seen in tests when reload time is "hacked" to 15 seconds.
Alternative is sending SIGUSR1 signal, but this seems to reload all modules, drop caches and is quite heavy.

Best way of improving this would be to watch on inotify on config files and apply changes as soon as they appear i n zone file.
Easier way, that can be implemented faster, is to expose reload time in plugin setup.

### PR DESCRIPTION
**Pull request by marqc**
Resolves #2108 

### 1. Why is this pull request needed and what does it do?

File plugin setup option to define reload interval

### 2. Which issues (if any) are related?

#2108 

### 3. Which documentation changes (if any) need to be made?

provided in PR in README.md file

### CODE REVIEW

File: plugin/file/reload.go [hunk 1]
```diff
@@ -5,15 +5,14 @@ import (
...
 
-// TickTime is the default time we use to reload zone. Exported to be tweaked in tests.
-var TickTime = 1 * time.Minute
+// Tick every second. Handler checks if reloadInterval has been reached.
+var TickTime = 1 * time.Second
 
 // Reload reloads a zone when it is changed on disk. If z.NoRoload is true, no reloading will be done.
 func (z *Zone) Reload() error {
-	if z.NoReload {
+	if z.NoReload || z.ReloadInterval == 0 {
```
Comment (miekg) [inline plugin/file/reload.go]: drop z.NoReload and make no_reload an alias for "reload 0"?

**Comment by miekg** [comment]
Could you make the same change to *auto*? That shares some of the setup code.

**Comment by miekg** [comment]
travis crashes in the weirdest places:
~~~
--- PASS: TestAutoNonExistentZone (0.00s)
panic: sync: negative WaitGroup counter
goroutine 2003 [running]:
sync.(*WaitGroup).Add(0xc00057f770, 0xffffffffffffffff)
	/home/travis/.gimme/versions/go1.11.linux.amd64/src/sync/waitgroup.go:74 +0x20d
sync.(*WaitGroup).Done(0xc00057f770)
	/home/travis/.gimme/versions/go1.11.linux.amd64/src/sync/waitgroup.go:99 +0x42
github.com/coredns/coredns/core/dnsserver.(*Server).Stop.func1(0xc00057f740, 0xc000adb380)
	/home/travis/gopath/src/github.com/coredns/coredns/core/dnsserver/server.go:167 +0x44
created by github.com/coredns/coredns/core/dnsserver.(*Server).Stop
	/home/travis/gopath/src/github.com/coredns/coredns/core/dnsserver/server.go:166 +0x8d
FAIL	github.com/coredns/coredns/test	31.907s
~~~

**Comment by marqc** [comment]
I applied same changes to auto plugin, removed noReload property from Zone as suugested and fixed tests based on short reload hack.

**Comment by marqc** [comment]
@miekg Can you review changes? Do you have more comments?

**Comment by miekg** [comment]
Totally forgot about this. Thanks for the ping. I'll take a look.

**Review by miekg — APPROVED** [review]

**Pull request merged**